pandas/_libs/lib.pyx

```diff
@@ -703,6 +703,10 @@ cpdef ndarray[object] ensure_string_array(
     if copy and result is arr:
         result = result.copy()
 
+    if util.is_array(arr) and issubclass(arr.dtype.type, np.str_):
```

Comment:
Should the `np.aasarray` call above guarantee `util.is_array(arr)` is always true?

Reply:
yep, you're right. I've removed the is_array check. thanks!